cmd/influxd/run.go

```diff
@@ -567,17 +569,19 @@ func (cmd *RunCommand) openServer(joinURLs []url.URL) *influxdb.Server {
 	if err := s.Open(cmd.config.Data.Dir, c); err != nil {
 		log.Fatalf("failed to open data node: %v", err.Error())
 	}
-	log.Printf("data node opened at %s", cmd.config.Data.Dir)
+	log.Printf("data node(%d) opened at %s", s.ID(), cmd.config.Data.Dir)
 
-	dataNodeIndex := s.Index()
-	if dataNodeIndex == 0 {
+	// Give brokers time to elect a leader if entire cluster is being restarted.
+	time.Sleep(1 * time.Second)
+
+	if s.ID() == 0 && s.Index() == 0 {
 		if len(joinURLs) > 0 {
 			joinServer(s, cmd.config.ClusterURL(), joinURLs)
 			return s
 		}
 
 		if err := s.Initialize(cmd.config.ClusterURL()); err != nil {
-			log.Fatalf("server initialization error: %s", err)
+			log.Fatalf("server initialization error(0): %s", err)
```

Comment:
OK, you could be right. It would still be better to see clearer log messages to help the next guy. E.g. `server initialization error during .....` -- I presume the software is in different states depending at the different points.